### PR TITLE
Automatically wrap with a Monitor when possible

### LINKED ISSUE #232: [Bug] Use of gym.make() stops "rollout/" data from being printed
### 🐛 Bug

If `gym.make` is used to define the environment the rollout data is not printed.

### To Reproduce

Example where `gym.make()` is used:

```python
import gym
from stable_baselines3 import SAC
env = gym.make('Pendulum-v0')
model = SAC('MlpPolicy', env, verbose = 1)
model.learn(200, log_interval = 1)
```
Output is missing the `rollout/` data:
```
Using cpu device
Wrapping the env in a DummyVecEnv.
---------------------------------
| time/              |          |
|    episodes        | 1        |
|    fps             | 71       |
|    time_elapsed    | 2        |
|    total timesteps | 200      |
| train/             |          |
|    actor_loss      | 7.2      |
|    critic_loss     | 2.08     |
|    ent_coef        | 0.971    |
|    ent_coef_loss   | -0.0491  |
|    learning_rate   | 0.0003   |
|    n_updates       | 99       |
---------------------------------
```

### Expected behavior

Example where `gym.make()` is not used:

```python
import gym
from stable_baselines3 import SAC
model = SAC('MlpPolicy', 'Pendulum-v0', verbose = 1)
model.learn(200, log_interval = 1)
```
Output includes the `rollout/` data:
```
Using cpu device
Creating environment from the given name 'Pendulum-v0'
Wrapping the env in a DummyVecEnv.
----------------------------------
| rollout/           |           |
|    ep_len_mean     | 200       |
|    ep_rew_mean     | -1.33e+03 |
| time/              |           |
|    episodes        | 1         |
|    fps             | 57        |
|    time_elapsed    | 3         |
|    total timesteps | 200       |
| train/             |           |
|    actor_loss      | 7.34      |
|    critic_loss     | 1.02      |
|    ent_coef        | 0.971     |
|    ent_coef_loss   | -0.0488   |
|    learning_rate   | 0.0003    |
|    n_updates       | 99        |
----------------------------------
```
### System Info

Describe the characteristic of your environment:
 * pip install stable-baselines3
 * Python 3.6.8
 * PyTorch version 1.7.0
 * Gym version 0.17.3

### PR DESCRIPTION
**Pull request by araffin**
## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

closes #232 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: stable_baselines3/common/base_class.py [hunk 1]
```diff
@@ -170,8 +167,22 @@ def __init__(
...
+
+        :param env:
+        :param verbose:
+        :param monitor_wrapper: Whether to wrap the env in a ``Monitor`` when possible.
+        :return: The wrapped environment.
+        """
         if not isinstance(env, VecEnv):
+            if not is_wrapped(env, Monitor) and monitor_wrapper:
+                if verbose >= 1:
+                    print("Wrapping the env in with a `Monitor` wrapper")
```
Comment (Miffyli) [inline stable_baselines3/common/base_class.py:184]:
```suggestion
                    print("Wrapping the env with a `Monitor` wrapper")
```

File: stable_baselines3/common/base_class.py [hunk 2]
```diff
@@ -170,8 +167,22 @@ def __init__(
...
+        Wrap environment with the appropriate wrappers if needed.
+        For instance, to have a vectorized environment
+        or to re-order the image channels.
+
+        :param env:
+        :param verbose:
+        :param monitor_wrapper: Whether to wrap the env in a ``Monitor`` when possible.
+        :return: The wrapped environment.
+        """
         if not isinstance(env, VecEnv):
```
Comment (Miffyli) [inline stable_baselines3/common/base_class.py:181]: What if we have a vecenv but those do not have monitor wrappers? Hard to check for this case, but could also be hard to miss.

Comment (araffin) [reply stable_baselines3/common/base_class.py:181]: we would need to add a method to `VecEnv` to wrap it in all cases.

But that's why we rather provide `make_vec_env`.
This PR is mainly meant when people are using single Gym env (as in #232 )

**Review by Miffyli — APPROVED** [review]
LGTM!

**Pull request merged**